workflow/controller/dag.go

```diff
@@ -449,7 +449,7 @@ func (woc *wfOperationCtx) executeDAGTask(ctx context.Context, dagCtx *dagContex
 		}
 
 		// Release acquired lock completed task.
-		if tmpl != nil {
```

Comment:
I agree, it looks like this was overlooked in #9405.

Did you consider whether there should actually be a guard higher up in the code against continuing this far if `tmpl == nil`? It seems odd that we don't `return` if line 446 returns error, in symmetry with line 437.

Reply:
Yes, I think we should get the error at line 434, and if error is nil, then must`tmpl != nil`

```
_, tmpl, _, tmplErr := dagCtx.tmplCtx.ResolveTemplate(task)
if tmplErr != nil {
	woc.markNodeError(node.Name, tmplErr)
	return
}
```